src/hotspot/share/gc/g1/heapRegionSet.cpp

```diff
@@ -227,40 +227,47 @@ void FreeRegionList::add_ordered(FreeRegionList* from_list) {
   add_list_common_end(from_list);
 }
 
+#ifdef ASSERT
+void FreeRegionList::verify_region_to_remove(HeapRegion* curr, HeapRegion* next) {
+  assert_free_region_list(_head != next, "invariant");
+  if (next != NULL) {
+    assert_free_region_list(next->prev() != NULL, "invariant");
```

Comment:
This assert could be next->prev() == curr, or am I missing some case?

Reply:
Thanks for pointing out, changed as you suggested.